sound/soc/sof/loader.c

```diff
@@ -284,6 +284,12 @@ int snd_sof_run_firmware(struct snd_sof_dev *sdev)
 	init_waitqueue_head(&sdev->boot_wait);
 	sdev->boot_complete = false;
 
+	/* create fw_version debugfs to store boot version info */
+	if (sdev->first_boot)
+		snd_sof_debugfs_create_item(sdev, &sdev->fw_version,
+					    sizeof(sdev->fw_version),
+					    "fw_version");
```

Comment:
this function can fail and the return status is not checked.
Actually in most cases the code does not check if this function is successful so please submit a PR to clean this up. Either we check the status everywhere or we don't have a status.
Thanks!

Reply:
@plbossart 
Will send another PR to clear debugfs functions.

I also find some thing that related to `snd_sof_debugfs_create_item`
The file ops are binded with the pm options, but when the DSP panic or oops, the PM will fail and make debugfs open/read fail.